source/common/upstream/upstream_impl.h

```diff
@@ -238,6 +221,9 @@ class HostSetImpl : public HostSet {
   }
 
 private:
+  // Weight for a locality taking into account health status.
+  double effectiveLocalityWeight(uint32_t index) const;
```

Comment:
lazyreview: I'm sure we have coverage of this but fo we have functional unit testing? It'd be nice to have, even if we're just querying the scheduler or some such.

Reply:
We have this coverage from `HostSetImplLocalityTest`. I think these tests are fairly unit/functional level, but they do include the scheduler. LMK if you want me to make this method `public` and I can do the finer grained test.